File: 242 - Valid Anagram/python/ans.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t): return False
        
        m1 = dict()
        m2 = dict()

        for i, c in enumerate(s):
            if c in m1:
                m1[c] += 1
            else:
                m1[c] = 1
            if t[i] in m2:
                m2[t[i]] += 1
            else:
                m2[t[i]] = 1

        for k in m1:
            if m1[k] != m2[k]:
                return False

        return True

File: 242 - Valid Anagram/cpp/ans.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.size() != t.size()) return false;
        unordered_map<char, int> m1;
        unordered_map<char, int> m2;
        for (int i = 0; i < s.size(); i++) {
            if (m1.find(s[i]) == m1.end()) {
                m1[s[i]] = 1;
            } else {
                m1[s[i]]++;
            }
            if (m2.find(t[i]) == m2.end()) {
                m2[t[i]] = 1;
            } else {
                m2[t[i]]++;
            }
        }
        
        return m1 == m2;
    }
};